examples/stress_tests/many_components.rs

```diff
@@ -0,0 +1,196 @@
+            }
+        }
+    }
+}
+
+fn stress_test(num_entities: u32, num_components: u32, num_systems: u32) {
+    let mut rng = ChaCha8Rng::seed_from_u64(42);
+    let mut app = App::default();
+    let world = app.world_mut();
+
+    // register a bunch of components
+    let component_ids: Vec<ComponentId> = (1..=num_components)
+        .map(|i| {
+            world.register_component_with_descriptor(
+                #[allow(unsafe_code)]
+                // SAFETY:
+                // we don't implement a drop function
+                // u8 is Sync and Send
+                unsafe {
+                    ComponentDescriptor::new_with_layout(
+                        format!("Component{}", i).to_string(),
+                        StorageType::Table,
+                        Layout::new::<u8>(),
+                        None,
+                        true, // is mutable
+                    )
+                },
+            )
+        })
+        .collect();
+
+    // fill the schedule with systems
+    let mut schedule = Schedule::new(Update);
+    for _ in 1..=num_systems {
+        let num_access_components = rng.gen_range(1..10);
+        let access_components: Vec<ComponentId> = component_ids
+            .choose_multiple(&mut rng, num_access_components)
+            .copied()
+            .collect();
+        let system = (QueryParamBuilder::new(|builder| {
+            for &access_component in &access_components {
+                if rand::random::<bool>() {
+                    builder.mut_id(access_component);
+                } else {
+                    builder.ref_id(access_component);
+                }
+            }
+        }),)
+            .build_state(world)
+            .build_any_system(base_system);
+        schedule.add_systems((move || access_components.clone()).pipe(system));
+    }
+
+    // spawn a bunch of entities
+    for _ in 1..=num_entities {
+        let num_components = rng.gen_range(1..10);
+        let components = component_ids.choose_multiple(&mut rng, num_components);
+
+        let mut entity = world.spawn_empty();
+        for &component_id in components {
+            let value: u8 = rng.gen_range(0..255);
+            OwningPtr::make(value, |ptr| {
+                #[allow(unsafe_code)]
+                // SAFETY:
+                // component_id is from the same world
+                // value is u8, so ptr is a valid reference for component_id
+                unsafe {
+                    entity.insert_by_id(component_id, ptr);
+                }
+            });
+        }
+    }
+
+    // overwrite Update schedule in the app
+    app.add_schedule(schedule);
+    app.add_plugins(MinimalPlugins)
+        .add_plugins(DiagnosticsPlugin)
+        .add_plugins(LogPlugin::default())
+        .add_plugins(FrameTimeDiagnosticsPlugin)
+        .add_plugins(LogDiagnosticsPlugin::filtered(vec![DiagnosticPath::new(
+            "fps",
+        )]));
+    app.run();
+}
+
+#[allow(missing_docs)]
```

Comment:
```suggestion
#[expect(missing_docs)]
```

(Nit) I largely prefer `#[expect(...)]`, since it emits a warning when the lint goes away.